sdk/python/kubeflow/training/api/py_torch_job_client.py

```diff
@@ -0,0 +1,385 @@
+                           expected_condition,
+                           namespace=None,
+                           timeout_seconds=600,
+                           polling_interval=30,
+                           status_callback=None):
+        """Waits until any of the specified conditions occur.
+
+        :param name: Name of the job.
+        :param expected_condition: A list of conditions. Function waits until any of the
+               supplied conditions is reached.
+        :param namespace: defaults to current or default namespace.
+        :param timeout_seconds: How long to wait for the job.
+        :param polling_interval: How often to poll for the status of the job.
+        :param status_callback: (Optional): Callable. If supplied this callable is
+               invoked after we poll the job. Callable takes a single argument which
+               is the job.
+        :return: Object: PyTorchJob
+        """
+
+        if namespace is None:
+            namespace = utils.get_default_target_namespace()
+
+        for _ in range(round(timeout_seconds / polling_interval)):
+
+            pytorchjob = None
+            pytorchjob = self.get(name, namespace=namespace)
+
+            if pytorchjob:
+                if status_callback:
+                    status_callback(pytorchjob)
+
+                # If we poll the CRD quick enough status won't have been set yet.
+                conditions = pytorchjob.get("status", {}).get("conditions", [])
+                # Conditions might have a value of None in status.
+                conditions = conditions or []
+                for c in conditions:
+                    if c.get("type", "") in expected_condition:
+                        return pytorchjob
+
+            time.sleep(polling_interval)
+
+        raise RuntimeError(
+            "Timeout waiting for PyTorchJob {0} in namespace {1} to enter one of the "
+            "conditions {2}.".format(name, namespace, expected_condition), pytorchjob)
+
+    def get_job_status(self, name, namespace=None):
+        """Returns PyTorchJob status, such as Running, Failed or Succeeded.
+
+        :param name: The PyTorchJob name.
+        :param namespace: defaults to current or default namespace.
+        :return: str: PyTorchJob status
+        """
+        if namespace is None:
+            namespace = utils.get_default_target_namespace()
+
+        pytorchjob = self.get(name, namespace=namespace)
+        last_condition = pytorchjob.get("status", {}).get("conditions", [])[-1]
+        return last_condition.get("type", "")
+
+    def is_job_running(self, name, namespace=None):
+        """Returns true if the PyTorchJob running; false otherwise.
+
+        :param name: The PyTorchJob name.
+        :param namespace: defaults to current or default namespace.
+        :return: True or False
+        """
+        pytorchjob_status = self.get_job_status(name, namespace=namespace)
+        return pytorchjob_status.lower() == "running"
```

Comment:
Move this status to [constants](https://github.com/kubeflow/tf-operator/blob/876a3e053d3e21002ff69a465362712b3c635b80/sdk/python/kubeflow/training/constants/constants.py) ?